addon/components/tinymce-editor.ts

```diff
@@ -0,0 +1,301 @@
+     * @link https://www.tiny.cloud/docs/plugins/opensource/table/
+     */
+    TABLE = 'table',
+    /**
+     * The template plugin adds support for custom templates. It also adds a menu item Insert template under the Insert menu and a toolbar button.
+     * @memberof TinymceEditorPlugins
+     * @link https://www.tiny.cloud/docs/plugins/opensource/template/
+     */
+    TEMPLATE = 'template',
+    /**
+     * The Text Pattern plugin matches special patterns in the text and applies formats, replaces text, or executes commands on these patterns
+     * @memberof TinymceEditorPlugins
+     * @link https://www.tiny.cloud/docs/plugins/opensource/textpattern/
+     */
+    TEXT_PATTERN = 'textpattern',
+    /**
+     * The toc plugin will generate basic Table of Contents and insert it into the editor at the current cursor position. Items for the table will be taken from the headers found in the content.
+     * @memberof TinymceEditorPlugins
+     * @link https://www.tiny.cloud/docs/plugins/opensource/toc/
+     */
+    TABLE_OF_CONTENTS = 'toc',
+    /**
+     * This plugin allows a user to see block level elements in the editable area. It is similar to WYSIWYG hidden character functionality, but at block level. It also adds a toolbar button and a menu item Show blocks under the View menu dropdown.
+     * @memberof TinymceEditorPlugins
+     * @link https://www.tiny.cloud/docs/plugins/opensource/visualblocks/
+     */
+    VISUAL_BLOCKS = 'visualblocks',
+    /**
+     * This plugin adds the ability to see invisible characters like &nbsp; displayed in the editable area. It also adds a toolbar control and a menu item Show invisible characters under the View menu.
+     * @memberof TinymceEditorPlugins
+     * @link https://www.tiny.cloud/docs/plugins/opensource/visualchars/
+     */
+    VISUAL_CHARACTERS = 'visualchars',
+    /**
+     * The Word Count plugin adds the functionality for counting words to the TinyMCE editor by placing a counter on the right edge of the status bar. Clicking Word Count in the status bar switches between counting words and characters. A dialog box with both word and character counts can be opened using the menu item situated in the Tools drop-down, or the toolbar button.
+     * @memberof TinymceEditorPlugins
+     * @link https://www.tiny.cloud/docs/plugins/opensource/wordcount/
+     */
+    WORD_COUNT = 'wordcount'
+}
+
+export interface TinymceEditorArgs extends RawEditorSettings {
+    plugins: TinymceEditorPlugins[];
+    onRender?: (event: any) => void;
+}
+
+export default class TinymceEditor extends Component<TinymceEditorArgs> {
+    get selector() {
+        return this.args.selector ?? `#${this.areaId}`;
+    }
+
+    get areaId() {
+        return guidFor(this);
+    }
+
+    @action
+    async initEditor() {
+        if (!window.tinymce) {
+            await import('tinymce').then((module) => module.default);
+        }
+        tinymce.baseURL = 'assets/';
+        await tinymce.init({
```

Comment:
if we can grab the editor instance from the return of this `init()`, we should probably also store a reference to it int he component, so that we can implement a `willDestroy()` method on the component and call `editor.destroy()`

https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#destroy